### PR TITLE
added types to new fields in `OptimizationConfig`

### LINKED ISSUE #1035: Untyped `dataclass` flieds in `OptimizationConfig`
### System Info

```shell
optimum_version: 1.8.3
```

### Who can help?

@regisss

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

```python
from optimum.onnxruntime import OptimizationConfig

OptimizationConfig(disable_group_norm_fusion=True)
```
Raises `TypeError: OptimizationConfig.__init__() got an unexpected keyword argument 'disable_group_norm_fusion'`

### Expected behavior

To set `disable_group_norm_fusion` to True.
The problem is in the [untyped fields](https://github.com/ericvsmith/dataclasses/issues/2) here:

https://github.com/huggingface/optimum/blob/e32ba52e4025692e1979cbbf0b4a55c02a3f2ac6/optimum/onnxruntime/configuration.py#L717-L722

Some of them have different names than in the doc:

https://github.com/huggingface/optimum/blob/e32ba52e4025692e1979cbbf0b4a55c02a3f2ac6/optimum/onnxruntime/configuration.py#L668-L679

### PR DESCRIPTION
**Pull request by IlyasMoutawwakil**
# What does this PR do?

Fixes #1035 

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Comment by IlyasMoutawwakil** [comment]
Because of confusion between the docs and default values, I disabled all opts for now.

**Review by regisss — APPROVED** [review]
LGTM!
Some tests failed but they are not related to this PR. Let's still wait a bit to solve this before merging.

**Comment by regisss** [comment]
@IlyasMoutawwakil The style check failed, I hadn't noticed it. Can you please run the following commands from the root of your Optimum repo?
```bash
pip install .[quality]
make style
```

**Comment by regisss** [comment]
@IlyasMoutawwakil Let's rebase the branch on the current main and all tests should pass :slightly_smiling_face:

**Comment by IlyasMoutawwakil** [comment]
@regisss I guess one test got cancelled after taking too long

File: optimum/onnxruntime/configuration.py [hunk 1]
```diff
@@ -715,11 +715,11 @@ class OptimizationConfig:
...
-    use_multi_head_attention = False
-    enable_gemm_fast_gelu_fusion = False
-    use_raw_attention_mask = False
-    disable_group_norm_fusion = True
-    disable_packed_kv = True
+    use_multi_head_attention: bool = False
+    enable_gemm_fast_gelu_fusion: bool = False
+    use_raw_attention_mask: bool = False
+    disable_group_norm_fusion: bool = False
+    disable_packed_kv: bool = False
```
Comment (regisss) [inline optimum/onnxruntime/configuration.py:722]: Reading the docstring of the class again, I think these two should be set to `True` by default (i.e. these optimizations are not done by default).

**Comment by regisss** [comment]
Okay, this test took 5 hours to complete: https://github.com/huggingface/optimum/blob/d29e5823bf976c84f9807a50fda9488a0ed54e85/tests/onnxruntime/test_utils.py#L21

But with Python 3.8 it got completed almost instantly. Very weird. I'll keep investigating.

**Comment by regisss** [comment]
These tests seem a bit flaky, but they all pass in the end.

**Pull request merged**